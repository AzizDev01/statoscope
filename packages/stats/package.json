{
  "name": "@statoscope/stats",
  "version": "5.7.0",
  "description": "Statoscope stats spec",
  "types": "./spec/index.d.ts",
  "scripts": {},
  "repository": "https://github.com/statoscope/statoscope/",
  "author": {
    "name": "Sergey Melyukov",
    "url": "https://twitter.com/smelukov"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "spec/*"
  ],
  "license": "MIT",
  "homepage": "https://github.com/statoscope/statoscope/"
}
