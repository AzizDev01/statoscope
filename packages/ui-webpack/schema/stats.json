{
  "definitions": {
    "Entrypoints": {
      "type": "object",
      "minProperties": 1,
      "additionalProperties": {
        "type": "object",
        "assets": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "chunks": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      }
    },
    "Chunks": {
      "type": "array",
      "minLength": 1,
      "items": {
        "$ref": "#/definitions/Chunk"
      }
    },
    "Chunk": {
      "type": "object",
      "properties": {
        "id": {
          "anyOf": [
            {
              "type": "number"
            },
            {
              "type": "string"
            }
          ]
        },
        "size": {
          "type": "number"
        },
        "modules": {
          "$ref": "#/definitions/Modules"
        },
        "names": {
          "type": "array",
          "items": {
            "anyOf": [
              {
                "type": "number"
              },
              {
                "type": "string"
              }
            ]
          }
        },
        "children": {
          "type": "array",
          "items": {
            "anyOf": [
              {
                "type": "number"
              },
              {
                "type": "string"
              }
            ]
          }
        },
        "files": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      }
    },
    "Assets": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Asset"
      }
    },
    "Asset": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        },
        "size": {
          "type": "number"
        },
        "chunks": {
          "type": "array",
          "items": {
            "anyOf": [
              {
                "type": "number"
              },
              {
                "type": "string"
              }
            ]
          }
        }
      }
    },
    "Modules": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Module"
      }
    },
    "Module": {
      "minLength": 1,
      "type": "object",
      "properties": {
        "id": {
          "oneOf": [
            {
              "enum": [
                null
              ]
            },
            {
              "type": "string"
            },
            {
              "type": "number"
            }
          ]
        },
        "identifier": {
          "type": "string"
        },
        "name": {
          "type": "string"
        },
        "size": {
          "type": "number"
        },
        "reasons": {
          "type": "array"
        }
      }
    }
  },
  "type": "object",
  "properties": {
    "version": {
      "type": "string"
    },
    "hash": {
      "type": "string"
    },
    "entrypoints": {
      "$ref": "#/definitions/Entrypoints"
    },
    "chunks": {
      "$ref": "#/definitions/Chunks"
    },
    "assets": {
      "$ref": "#/definitions/Assets"
    }
  }
}
