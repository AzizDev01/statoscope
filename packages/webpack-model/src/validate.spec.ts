import stats from '../../../test/bundles/simple/stats-prod.json';
import validateStats from './validate';

test('should work', () => {
  expect(validateStats(stats).result).toBeTruthy();
  // @ts-ignore
  expect(validateStats(stats).errors).toBeUndefined();
  // @ts-ignore
  stats.chunks[0].modules[0].id = true;
  expect(validateStats(stats)).toMatchSnapshot();
});
