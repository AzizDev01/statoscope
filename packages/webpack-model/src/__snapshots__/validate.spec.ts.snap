// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Object {
  "errors": Array [
    Object {
      "instancePath": "/chunks/0/modules/0/id",
      "keyword": "enum",
      "message": "must be equal to one of the allowed values",
      "params": Object {
        "allowedValues": Array [
          null,
        ],
      },
      "schemaPath": "#/definitions/Module/properties/id/oneOf/0/enum",
    },
    Object {
      "instancePath": "/chunks/0/modules/0/id",
      "keyword": "type",
      "message": "must be string",
      "params": Object {
        "type": "string",
      },
      "schemaPath": "#/definitions/Module/properties/id/oneOf/1/type",
    },
    Object {
      "instancePath": "/chunks/0/modules/0/id",
      "keyword": "type",
      "message": "must be number",
      "params": Object {
        "type": "number",
      },
      "schemaPath": "#/definitions/Module/properties/id/oneOf/2/type",
    },
    Object {
      "instancePath": "/chunks/0/modules/0/id",
      "keyword": "oneOf",
      "message": "must match exactly one schema in oneOf",
      "params": Object {
        "passingSchemas": null,
      },
      "schemaPath": "#/definitions/Module/properties/id/oneOf",
    },
    Object {
      "instancePath": "",
      "keyword": "anyOf",
      "message": "must match a schema in anyOf",
      "params": Object {},
      "schemaPath": "#/anyOf",
    },
  ],
  "result": false,
}
`;
